src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -50,28 +75,41 @@ public enum LuceneStorage
         // Holds the instance for the IndexSearcher
         internal IndexSearcher Searcher;
 
-        internal LuceneSearchUtility(DynamoModel model)
+        /// <summary>
+        /// Default constructor for LuceneSearchUtility, it will use the default storage type - RAM
+        /// </summary>
+        internal LuceneSearchUtility()
+        {
+            startConfig = new LuceneStartConfig();
+        }
+
+        /// <summary>
+        /// Constructor for LuceneSearchUtility, it will use the storage type passed as parameter
+        /// </summary>
+        /// <param name="config"></param>
+        internal LuceneSearchUtility(LuceneStartConfig config)
         {
-            dynamoModel = model;
+            startConfig = config;
+            if (DynamoModel.IsTestMode)
+            {
+                // If under test mode, switch to the default storage type - RAM
+                startConfig = new LuceneStartConfig();
+            }
         }
 
         /// <summary>
         /// Initialize Lucene config file writer.
         /// </summary>
-        internal void InitializeLuceneConfig(string dirName, LuceneStorage storageType = LuceneStorage.FILE_SYSTEM)
+        internal void InitializeLuceneConfig()
         {
             addedFields = new List<string>();
 
             DirectoryInfo luceneUserDataFolder;
-            var userDataDir = new DirectoryInfo(dynamoModel.PathManager.UserDataDirectory);
+            var userDataDir = new DirectoryInfo(PathManager.Instance.UserDataDirectory);
```

Comment:
This is neat for now, I am getting more curious if it is DynamoModel we need to make it Singleton or individual classes. I guess opening up the whole DynamoModel will be too much, which is why I proposed about each class needed.